### DISCUSSION TITLE
Issue with schema validation on OsgiGraphQLHttpServlet v15

### DISCUSSION THREAD

**Discussion by gflores-jahia** [graphql-java-kickstart/graphql-java-servlet#555] [Q&A] [answered]
Hello,

I'm having issue building/validating a graphql schema (specifically [this rule](https://github.com/graphql-java/graphql-java/blob/v20.2/src/main/java/graphql/schema/validation/SchemaValidator.java#L25)) because of missing directives when `OsgiGraphQLHttpServlet.bindProvider(GraphQLProvider)` ([code link](https://github.com/graphql-java-kickstart/graphql-java-servlet/blob/master/graphql-java-servlet/src/main/java/graphql/kickstart/servlet/OsgiGraphQLHttpServlet.java#L81)) is invoked.

The problem is I can't seem to figure out how/where to inject directive definitions when schema is being created (in doUpdateSchema() [here](https://github.com/graphql-java-kickstart/graphql-java-servlet/blob/master/graphql-java-servlet/src/main/java/graphql/kickstart/servlet/OsgiSchemaBuilder.java#L101)) i.e. `Builder.additionalDirective` is not populated during schema update. Or do we need to maybe provide a directive provider interface? I can create a PR if needed.

Any help/pointers would be appreciated, thanks in advance.

**Comment by federicorispo** [comment]
Hi @gflores-jahia,
I made a quick check in the codebase and, as you pointed out, there is no way to add/use directives on a Osgi servlet so a specific provider is needed.

Thanks for your contribution with [this PR](https://github.com/graphql-java-kickstart/graphql-java-servlet/pull/557) :tada:

**Comment by federicorispo** [comment, marked as answer]
@gflores-jahia I merged your PR on master and you can find the fix in the 15.2.1-SNAPSHOT version:

```xml
      <dependency>
        <groupId>com.graphql-java-kickstart</groupId>
        <artifactId>graphql-java-servlet-javax</artifactId>
        <version>15.2.1-SNAPSHOT</version>
      </dependency>
```

the specification of the following repository is required:
```xml
    <repository>
      <id>oss.sonatype.org-snapshot</id>
      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </repository>
```

**Comment by gflores-jahia** [comment]
Thanks!
